File: icons/generate-icons.py
#!/usr/bin/env python3
"""Render PNG PWA icons from icons/icon.svg using resvg-js."""

import subprocess
import sys
from pathlib import Path

ICON_DIR = Path(__file__).parent
SVG = ICON_DIR / 'icon.svg'


def main():
    if not SVG.exists():
        print(f'Missing {SVG}', file=sys.stderr)
        sys.exit(1)

    for size in (192, 512):
        output = ICON_DIR / f'icon-{size}.png'
        subprocess.run(
            [
                'npx',
                '--yes',
                '@resvg/resvg-js-cli',
                str(SVG),
                str(output),
                '--fit-width',
                str(size),
            ],
            check=True,
        )
        print(f'wrote {output}')


if __name__ == '__main__':
    main()

File: scripts/generate-icons.mjs
import fs from 'node:fs';
import path from 'node:path';
import { fileURLToPath } from 'node:url';
import { Resvg } from '@resvg/resvg-js';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const repoRoot = path.resolve(__dirname, '..');
const iconDir = path.join(repoRoot, 'publish', 'icons');
const svgPath = path.join(iconDir, 'icon.svg');

function main() {
    if (!fs.existsSync(svgPath)) {
        console.error(`Missing ${svgPath}`);
        process.exit(1);
    }

    const svgSource = fs.readFileSync(svgPath, 'utf8');

    for (const size of [192, 512]) {
        const resvg = new Resvg(svgSource, {
            fitTo: {
                mode: 'width',
                value: size,
            },
        });
        const output = path.join(iconDir, `icon-${size}.png`);

        fs.writeFileSync(output, resvg.render().asPng());
        console.log(`wrote ${output}`);
    }
}

main();
